### DISCUSSION TITLE
Learning Management Systems – LMS (Canvas, BlackBoard, etc.)

### DISCUSSION THREAD

**Discussion by baptiste** [quarto-dev/quarto-cli#4924] [Feature Requests]
_(For your own sake I sincerely hope you have no idea what the title refers to (please feel free to stop reading now, you're better off not knowing))_

These Learning Management Systems (LMS) often come with an API ([here is Canvas'](https://canvas.instructure.com/doc/api/)), making it possible to generate content programmatically. Potentially, one could therefore author some content in Quarto, and upload it to the online system programmatically.

I've fixed a couple of glitches in the [{rcanvas} package](https://github.com/baptiste/rcanvas), and my initial experiments suggest that it can work reasonably well.

I'm posting this here in case others are having the same idea, so we could discuss ways to improve the integration. For example, 

- something like the exams package could allow the creation of online quizzes, integrated with the platform, etc. 
- Another idea for improvement would be to make it easier to refer to assets (files uploaded to the system are given an ID, and it would be nice if quarto could use this file organisation for images and other assets. 
- Canvas seems to use some kind of web framework (judging by the css classes, I haven't dug very deep), and so it could be nice to have quarto emit HTML with those css attributes.

**Comment by lexnederbragt** [comment]
Great suggestion! I already do something along these lines. I use different makeup language ([DocOnce] (https://doconce.github.io/doconce/doc/pub/manual/manual.html), a tool with rich functionality but a very small user base) for writing the Canvas pages in a markup language, and converting these to plain html. For adding/updating these pages, or files, to Canvas, I have written a Python-base command-line program that uses the [Python CanvasAPI](https://canvasapi.readthedocs.io/en/stable/getting-started.html), called [canvasCourseCli](https://github.com/lexnederbragt/canvasCourseCli). (Finally, I automate the process using (snake)make).

It would already be possible to use Quarto for writing the pages, and converting them to html. For the next course I may develop, this will become my workflow (again, in combination with canvasCourseCli).

It would be great if this workflow, using Quarto, could become very smooth, for example along the lines you sketch!
